feature/security/gnsi/certz/tests/client_certificates/client_certificates_test.go

```diff
@@ -0,0 +1,254 @@
+			clientCertzFile:  "test_data/ca-1000/client-ecdsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-1000/client-ecdsa-a-key.pem",
+			mismatch:         false,
+		},
+		{
+			desc:             "Certz1.2:Load the rsa trust_bundle from ca-02 with mismatching key type rsa client certificate from ca-01",
+			serverCertzFile:  "test_data/ca-01/server-rsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-01/server-rsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-02/trust_bundle_02_rsa_a.pem",
+			clientCertzFile:  "test_data/ca-01/client-rsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-01/client-rsa-a-key.pem",
+			mismatch:         true,
+		},
+		{
+			desc:             "Certz1.2:Load the ecdsa trust_bundle from ca-02 with mismatching key type ecdsa client certificate from ca-01",
+			serverCertzFile:  "test_data/ca-01/server-ecdsa-a-cert.pem",
+			serverKeyzFile:   "test_data/ca-01/server-ecdsa-a-key.pem",
+			trustBundlezFile: "test_data/ca-02/trust_bundle_02_ecdsa_a.pem",
+			clientCertzFile:  "test_data/ca-01/client-ecdsa-a-cert.pem",
+			clientKeyzFile:   "test_data/ca-01/client-ecdsa-a-key.pem",
+			mismatch:         true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.desc, func(t *testing.T) {
+
+			sc1, err := os.ReadFile(tc.serverCertzFile)
+			if err != nil {
+				t.Fatalf("Failed to read certificate: %v", err)
+			}
+			block, _ := pem.Decode(sc1)
+			if block == nil {
+				t.Fatalf("Failed to parse PEM block containing the public key.")
+			}
+			sCert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				t.Fatalf("Failed to parse certificate: %v", err)
+			}
+
+			san := sCert.DNSNames[0]
+
+			serverCert := setupService.CreateCertzChain(t, setupService.CertificateChainRequest{
+				RequestType:    setupService.EntityTypeCertificateChain,
+				ServerCertFile: tc.serverCertzFile,
+				ServerKeyFile:  tc.serverKeyzFile})
+
+			serverCertEntity := setupService.CreateCertzEntity(t, setupService.EntityTypeCertificateChain, &serverCert, "one")
+
+			trustCertChain := setupService.CreateCertChainFromTrustBundle(tc.trustBundlezFile)
+			trustBundleEntity := setupService.CreateCertzEntity(t, setupService.EntityTypeTrustBundle, trustCertChain, "two")
+
+			cert, err := tls.LoadX509KeyPair(tc.clientCertzFile, tc.clientKeyzFile)
+			if err != nil {
+				t.Fatalf("Failed to load  client cert: %v", err)
+			}
+
+			cacert := x509.NewCertPool()
+			cacertBytes, err := os.ReadFile(tc.trustBundlezFile)
+			if err != nil {
+				t.Fatalf("Failed to read ca bundle :%v", err)
+			}
+			if ok := cacert.AppendCertsFromPEM(cacertBytes); !ok {
+				t.Fatalf("Failed to parse %v", tc.trustBundlezFile)
+			}
+
+			certzClient := gnsiC.Certz()
+			success := setupService.CertzRotate(t, certzClient, testProfile, &serverCertEntity, &trustBundleEntity)
+			if !success {
+				t.Fatalf("%s:Certz rotation failed.", tc.desc)
+			}
+			t.Logf("%s:successfully completed certz rotation!", tc.desc)
+
+			if tc.mismatch != true {
```

Comment:
there's a 'do not if else' thing in golang... or that got hammered hard on me :( but...
you can just make this a switch:

  switch tc.mismatch {
  case true:
    stuff
   case false:
     other stuff
   }

Reply:
it also almost looks like you could either simplify as below, OR move the:
   result := .....
   
 outside the switch statement and just check mismatch to toss the error or not.